docs/merkletree/treetex/main.go

```diff
@@ -96,13 +99,14 @@ var (
 
 // nodeInfo represents the style to be applied to a tree node.
 type nodeInfo struct {
-	incProof     bool
-	incPath      bool
-	target       bool
-	perfectRoot  bool
-	ephemeral    bool
-	leaf         bool
-	rangeIndices []int
+	proof             bool
```

Comment:
This type mixes "global" node properties (e.g. `ephemeral` and `perfectRoot`) with query-dependent ones (proofs, ranges, etc). Consider storing a slice of styles/modifiers/classes instead of a bunch of bools for the query dependent properties - this would allow you to also have multiple inclusion proofs at the same time etc.

Reply:
That sounds like a good idea, but it's a job for another day.